stew/io2.nim

```diff
@@ -206,6 +230,15 @@ elif defined(posix):
       O_CLOEXEC = cint(0x1000000)
       F_NOCACHE = cint(48)
 
+  type
+    FlockStruct* {.importc: "struct flock", final, pure,
+                   header: "<fcntl.h>".} = object
+      ltype* {.importc: "l_type".}: cshort
+      lwhence* {.importc: "l_whence".}: cshort
+      start* {.importc: "l_start".}: int
+      length* {.importc: "l_len".}: int
+      pid* {.importc: "l_pid".}: Pid
```

Comment:
```suggestion
      ltype* {.importc: "l_type".}: cshort
      lwhence* {.importc: "l_whence".}: cshort
      start* {.importc: "l_start".}: Off
      length* {.importc: "l_len".}: Off
      pid* {.importc: "l_pid".}: Pid
```

https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/fcntl.h.html
start / length are technically an `off_t`

Reply:
It is actually not true, because:
```c
struct flock
  {
    short int l_type;   /* Type of lock: F_RDLCK, F_WRLCK, or F_UNLCK.  */
    short int l_whence; /* Where `l_start' is relative to (like `lseek').  */
#ifndef __USE_FILE_OFFSET64
    __off_t l_start;    /* Offset where the lock begins.  */
    __off_t l_len;      /* Size of the locked area; zero means until EOF.  */
#else
    __off64_t l_start;  /* Offset where the lock begins.  */
    __off64_t l_len;    /* Size of the locked area; zero means until EOF.  */
#endif
    __pid_t l_pid;      /* Process holding the lock.  */
  };
```

Reply:
So on 64bit platform `l_start` and `l_len` are `__off64_t` instead of `__off_t`